src/client/connection.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @hidden
+ */
+
+/**
+ * File contains vars used to establish service connection between client and provider.
+ *
+ * These are separated out from 'internal.ts' as including these from provider code will cause the provider to connect
+ * to itself.
+ *
+ * These types are a part of the client, but are not required by applications wishing to interact with the service.
+ * This file is excluded from the public-facing TypeScript documentation.
+ */
+import {EventEmitter} from 'events';
+import {ChannelClient} from 'openfin/_v2/api/interappbus/channel/client';
+
+import {APITopic, SERVICE_CHANNEL, SERVICE_IDENTITY, TopicPayloadMap, TopicResponseMap} from './internal';
+
+/**
+ * The version of the NPM package.
+ *
+ * Webpack replaces any instances of this constant with a hard-coded string at build time.
+ */
+declare const PACKAGE_VERSION: string;
+
+/**
+ * Defines all events that are fired by the service
+ */
+export type FDC3Event = any;  // tslint:disable-line:no-any
+
+/**
+ * The event emitter to emit events received from the service.  All addEventListeners will tap into this.
+ */
+export const eventEmitter = new EventEmitter();
+
+/**
+ * Promise to the channel object that allows us to connect to the client
+ */
+export let channelPromise: Promise<ChannelClient>;
+// Currently a runtime bug when provider connects to itself. Ideally the provider would never import a file
```

Comment:
Worth an explicit `Todo:` or a reference to a ticket?

Reply:
I'd say it's not strictly a TODO, but just an explanation for why we need that specific guard. I don't think there is any harm in keeping it there.